pkg/kn/commands/plugin/install.go

```diff
@@ -0,0 +1,114 @@
+// Copyright © 2024 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/config"
+)
+
+// NewPluginInstallCommand creates a new `kn plugin install` command
+func NewPluginInstallCommand(p *commands.KnParams) *cobra.Command {
+
+	var localPath string
+
+	cmd := &cobra.Command{
+		Use:     "install",
+		Short:   "Install a plugin from URL or local file",
+		Aliases: []string{"ls"},
+		Long: `Install a plugin from URL.or local filesystem.
+
+Plugin is installed from a provided URL to ${XDG_HOME}/kn/plugins.
+
+Current: ` + config.GlobalConfig.PluginsDir(),
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
maybe you can move the inline func to a named function. It's quite long, so it would enhance readability to have it separate.

Reply:
That name local function could return the `func` to be added here, so that the closure with `localPath` would work.

Reply:
Heh, you can always spot where I'm being lazy cleaning up my spaghetti code. :)